### ISSUE TITLE
Compilation issue

### ISSUE THREAD

**Issue by pzinn** [Macaulay2/M2#1271]
As requested, making an issue. not sure how to title it, since I haven't really looked into the source of the problem. anyway, starting from scratch on Fedora 32 
```
cat /proc/version 
Linux version 5.6.15-300.fc32.x86_64 (mockbuild@bkernel03.phx2.fedoraproject.org) (gcc version 10.1.1 20200507 (Red Hat 10.1.1-1) (GCC)) #1 SMP Fri May 29 14:23:59 UTC 2020
gcc --version
gcc (GCC) 10.1.1 20200507 (Red Hat 10.1.1-1)
```
compiling with `make`, I go as far as
```
make[3]: Entering directory '/home/pzinn/M2/M2/BUILD/fedora/libraries/mpsolve/build/mpsolve-3.1.8'
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh /home/pzinn/M2/M2/BUILD/fedora/libraries/mpsolve/build/mpsolve-3.1.8/missing aclocal-1.16 -I m4
 cd . && /bin/sh /home/pzinn/M2/M2/BUILD/fedora/libraries/mpsolve/build/mpsolve-3.1.8/missing automake-1.16 --gnu
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh /home/pzinn/M2/M2/BUILD/fedora/libraries/mpsolve/build/mpsolve-3.1.8/missing autoconf
/bin/sh ./config.status --recheck
running CONFIG_SHELL=/bin/sh /bin/sh ./configure --prefix=/home/pzinn/M2/M2/BUILD/fedora/usr-host AR=ar AS=as DLLTOOL=false OBJDUMP=objdump STRIP=strip CXXFLAGS=-std=gnu++14 -g3 -O2  -Wno-mismatched-tags -w CFLAGS=-std=gnu11 -g3 -O2 -w FFLAGS=-O2 FCFLAGS=-O2 TARGET_ARCH= CPPFLAGS=-I/home/pzinn/M2/M2/BUILD/fedora/../../include -I/home/pzinn/M2/M2/BUILD/fedora/include -I/home/pzinn/M2/M2/BUILD/fedora/usr-host/include -I/home/pzinn/M2/M2/BUILD/fedora/usr-host/include/cdd  -isystem /usr/include/libxml2 -isystem /usr/include/libxml2       -DNDEBUG  -I/usr/include/eigen3  CC=gcc CXX=g++ LDFLAGS=-L/home/pzinn/M2/M2/BUILD/fedora/usr-host/lib  -L/home/pzinn/M2/M2/BUILD/fedora/submodules/memtailor/.libs  -L/home/pzinn/M2/M2/BUILD/fedora/submodules/mathic/.libs  -L/home/pzinn/M2/M2/BUILD/fedora/submodules/mathicgb/.libs -g3 LOADLIBES= LDLIBS= --disable-shared --disable-examples --disable-ui --disable-documentation --cache-file=/dev/null PKG_CONFIG_PATH=/home/pzinn/M2/M2/BUILD/fedora/usr-host/lib/pkgconfig: --no-create --no-recursion
.
. [skipped]
.
checking for mpfr.h... yes
./configure: line 15645: syntax error near unexpected token `CHECK,'
./configure: line 15645: `  PKG_CHECK_MODULES(CHECK, check >= 0.9.4, have_check=yes, have_check=no)'
make[3]: *** [Makefile:449: config.status] Error 2
make[3]: Leaving directory '/home/pzinn/M2/M2/BUILD/fedora/libraries/mpsolve/build/mpsolve-3.1.8'
make[2]: *** [../Makefile.library:178: .compiled-3.1.8] Error 2
make[2]: Leaving directory '/home/pzinn/M2/M2/BUILD/fedora/libraries/mpsolve'
make[1]: *** [Makefile:7: all-in-mpsolve] Error 2
make[1]: Leaving directory '/home/pzinn/M2/M2/BUILD/fedora/libraries'
make: *** [GNUmakefile:211: all-in-libraries] Error 2
make: Leaving directory '/home/pzinn/M2/M2/BUILD/fedora'
```

**Comment by DanGrayson** [comment]
I saw the same error recently -- the problem is that the macro "PKG_CHECK_MODULES" is undefined.  It gets defined if you install `pkg-config`, called `pkgconf` on Fedora, by having its definition deposited in the file `/usr/share/aclocal/pkg.m4`.

I don't know what makes the definitions in that file get read!

I'll add that name to the INSTALL file in the Fedora section.

**Comment by DanGrayson** [comment]
See commit c79a887ed90f347a6ebdd11430e627b985a391c7, closing

**Comment by DanGrayson** [comment]
If that doesn't fix it, reopen.

**Comment by pzinn** [comment]
it did not work. I ended up manually copying `pkg.m4` from `/usr/share/aclocal` to the library m4 directory. that worked, but it's not great.

**Comment by pzinn** [comment]
in the meantime I got help from Mahrud and managed to compile (using cmake), so I removed my comment. maybe it was all my fault.

**Comment by DanGrayson** [comment]
I don't see how it can be your fault.  To test that, rebuild `mpsolve` from scratch by running
```
make clean
make
```
in `libraries/mpsolve` in the build tree.

> it did not work. I ended up manually copying pkg.m4 from /usr/share/aclocal to the library m4 directory. that worked, but it's not great.

I looked into it some more.  The program `aclocal` scans `configure.ac` looking for macros that might require definitions, finds the definitions, and inserts them into the file `aclocal.m4`, which gets used by `autoconf` when creating `configure`.  This has been done already by the authors of `mpsolve`, the library involved in your report, and the files `aclocal.m4` and `configure` are included in the tar file, `M2/BUILD/tarfiles/mpsolve-3.1.8.tar.gz`.  My makefile doesn't touch those files or rerun `autoreconf`, but a patch file is applied to `configure.ac`, and the makefile of mpsolve notices and reruns `automake`, which presumably reruns `aclocal`.

So what must be happening is that your `aclocal` is not looking in the directory `/usr/share/aclocal` for whatever reason.

Look in `/usr/bin/aclocal` -- it is a perl script, and these lines specify where it should look:
```
my @automake_includes = ('/usr/share/aclocal-' . $APIVERSION);
my @system_includes = ('/usr/share/aclocal');
```

Run a test by creating a file somewhere called `configure.ac` containing the single line `PKG_CHECK_MODULES()`, run `aclocal` there, and see whether the file `aclocal.m4` that results contains the definition.

**Comment by mahrud** [comment]
I'm not sure if this issue in particular is resolved. I wasn't able to reproduce it on fedora, and Paul is using cmake now, so I think it can be closed.

**Comment by DanGrayson** [comment]
I see now that another way this problem can arise is if `make get-automake` is run at the top of the source tree.  The problem then is that along with automake comes aclocal, and the one we build does not look in `/usr/share/aclocal`.  But long ago I also added `pkg.m4` to the things that get installed when `make get-tools` is run (see the file `M2/usr-build/pkg.m4`), so I've updated it and added it as a prerequisite for `get-automake`. 

That should improve things.
